ipn/ipnlocal/local.go

```diff
@@ -3335,18 +3346,6 @@ func (b *LocalBackend) TCPHandlerForDst(src, dst netip.AddrPort) (handler func(c
 	if dst.Port() == webClientPort && b.ShouldRunWebClient() {
 		return b.handleWebClientConn, opts
 	}
-	if dst.Port() == TailFSLocalPort {
```

Comment:
This is the main fix, don't intercept TailFS traffic from external connections to the local IP.